lib/codec_choose.c

```diff
@@ -49,6 +49,9 @@
 #ifndef bit_SSE42
 #define bit_SSE42 (1 << 20)
 #endif
+#ifndef bit_AVX
+#define bit_AVX (1 << 28)
+#endif
 
 #define bit_XSAVE_XRSTORE (1 << 27)
```

Comment:
The wikipedia states it is bit 28 for `Advanced Vector eXtensions` (modified case on `eXtensions` on purpose).
Bit 26 & 27 are `XSAVE/XRESTOR` & `XSAVE enabled by OS`.
The mechanism to check for `AVX` is the same as `AVX2` except for the `AVX` flag instead of `AVX2`. Meaning all the stuff around checking OS support, calling `xgetbv`is the same.
Yes, the OS can have an influence on `CPUID`. If I remember correctly (this could be checked in the [Intel® 64 and IA-32 Architectures Software Developer Manuals](http://www.intel.com/content/www/us/en/processors/architectures-software-developer-manuals.html)), when the OS/kernel sets-up the CPU, it configures specific registers telling it that it will use `XSAVE/XRESTORE` and supports saving `ymm` registers. The first reflects in `CPUID`and the latter in the call to `xgetbv`.